### ISSUE TITLE
`prefer-export-from` shouldn't report re-export of variable that is used in other statements

### ISSUE THREAD

**Issue by FloEdelmann** [sindresorhus/eslint-plugin-unicorn#1577]
```js
import { a } from './my-module.js';

export const exportA = a;
export const exportB = a.b;
```

The `prefer-export-from` rule warns about the first export and wants to change it to:

```js
import { a } from './my-module.js';


export const exportB = a.b;
export { a as exportA } from './my-module.js';
```

I think that this duplication is more confusing: `a` imported as `a`, but also re-exported with a different name.

I therefore suggest to ignore exports of variables that are used in other statements.

Real-life reproduction: [OpenLightingProject/open-fixture-library@`dependabot/npm_and_yarn/eslint-plugin-unicorn-38.0.1`/lib/schema-properties.js](https://github.com/OpenLightingProject/open-fixture-library/blob/dependabot/npm_and_yarn/eslint-plugin-unicorn-38.0.1/lib/schema-properties.js)

**Comment by fisker** [comment]
Check this comment and related comments https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1453#issuecomment-888294004

**Comment by fregante** [comment]
@fisker those comments seems _a bit_ reasonable with a `foo` module, but re-exporting other npm modules is rare and not really advised, especially `export default from`

The example should use local files:

```js
import {MyTinyForm} from '../components/forms/tiny.js';
export {MyTinyForm} from '../components/forms/tiny.js';

console.log('Registered', MyTinyForm.getName());
```

… and that's awful compared to

```js
import {MyTinyForm} from '../components/forms/tiny.js';
export {MyTinyForm};

console.log('Registered', MyTinyForm.getName());
```

Even OP’s example shows that the rule does not suggest cleaner code.

**Comment by fisker** [comment]
A `ignoredUsedVariables` option?

**Comment by fisker** [comment]
@sindresorhus ^

**Comment by sindresorhus** [comment]
Sure. `ignoredUsedVariables` => `ignoreUsedVariables`
